core/node/rpc/archiver.go

```diff
@@ -181,17 +181,42 @@ func (a *Archiver) ArchiveStream(ctx context.Context, stream *ArchiveStream) err
 	}
 
 	for mbsInDb < mbsInContract {
+
 		toBlock := min(mbsInDb+int64(a.config.GetReadMiniblocksSize()), mbsInContract)
-		resp, err := stub.GetMiniblocks(
-			ctx,
-			connect.NewRequest(&GetMiniblocksRequest{
-				StreamId:      stream.streamId[:],
-				FromInclusive: mbsInDb,
-				ToExclusive:   toBlock,
-			}),
-		)
+
+		retries := 3
+		var resp *connect.Response[GetMiniblocksResponse]
+		for retries > 0 {
+			resp, err = stub.GetMiniblocks(
+				ctx,
+				connect.NewRequest(&GetMiniblocksRequest{
+					StreamId:      stream.streamId[:],
+					FromInclusive: mbsInDb,
+					ToExclusive:   toBlock,
+				}),
+			)
+			if err != nil {
+				log.Warn(
+					"Error when calling GetMiniblocks on server",
+					"error",
+					err,
+					"streamId",
+					stream.streamId,
+					"retries",
+					retries,
+				)
+				stream.nodes.AdvanceStickyPeer(nodeAddr)
+
+				if AsRiverError(err).Code != Err_NOT_FOUND {
+					return err
+				}
+			}
+			// Wait half a second before retrying
+			time.Sleep(time.Millisecond * 500)
```

Comment:
I'm not deeply familiar with the archive code. But is this long enough? And if after 3 attempts it still fails is it rescheduled at some time in the future?

Reply:
It is actually rescheduled for 1 second later by this loop, I'm refactoring now.